Harvest.Web/Controllers/Api/ProjectController.cs

```diff
@@ -185,5 +186,115 @@ public async Task<IActionResult> RefreshTotal(int projectId)
+        {
+            postModel.Project = new Project();
+            postModel.Project.Name = "Fake One 3";
+            postModel.Project.Crop = "Special Services";
+            postModel.Project.CropType = "Row";
+            postModel.Project.Start = DateTime.Now;
+            postModel.Project.End = DateTime.Now.AddMonths(1);
+            postModel.Project.Requirements = "Fake Requirements";
+
+            postModel.Accounts = new Account[2];
+            postModel.Accounts[0] = new Account();
+            postModel.Accounts[1] = new Account();
+            postModel.Accounts[0].Percentage = 75m;
+            postModel.Accounts[0].Name = "CrU 1";
+            postModel.Accounts[0].Number = "3-CRU9033";
+
+            postModel.Accounts[1].Percentage = 25m;
+            postModel.Accounts[1].Name = "CrU 2";
+            postModel.Accounts[1].Number = "3-CRUEQIP";
+
+
+
+
+            var currentUser = await _userService.GetCurrentUser();
+            var newProject = new Project
+            {
+                Name = postModel.Project.Name,
+                Crop = postModel.Project.Crop,
+                CropType = postModel.Project.CropType,
+                Start = postModel.Project.Start, //Start and stop just now and a month from now?
+                End = postModel.Project.End,
+                CreatedOn = DateTime.UtcNow,
+                CreatedById = currentUser.Id,
+                IsActive = true,
+                IsApproved = true,
+                Requirements = postModel.Project.Requirements,
+                PrincipalInvestigatorId = currentUser.Id,
+            };
+            newProject.UpdateStatus(Project.Statuses.Active);
+
+            newProject.Acres = 0;
+            newProject.ChargedTotal = 0;
+            newProject.QuoteTotal = postModel.Expenses.Select(a => a.Total).Sum();
+           
+
+            try
+            {
+                await _dbContext.Projects.AddAsync(newProject);
+                await _dbContext.SaveChangesAsync();
+            }
+            catch (Exception ex)
+            {
+                var xxx = ex.Message;
+            }
+
+
+            newProject.Expenses =  new List<Expense>();
+            //Save project first if projectId needed. Wrap in a transaction
+            var percentage = 0.0m;
+
+            foreach (var account in postModel.Accounts)
+            {
+                // Accounts will be auto-approved by quote approver
+                account.ApprovedById = currentUser.Id;
+                account.ProjectId= newProject.Id;
+                account.ApprovedOn = DateTime.UtcNow;
+                percentage += account.Percentage;
+                if (account.Percentage < 0)
+                {
+                    return BadRequest("Negative Percentage Detected");
+                }
+                newProject.Accounts.Add(account); //Don't need to specify projectId if I add it to the project?
```

Comment:
I believe yes, as long as you are adding to the array you don't need to set projectId, and thus you don't need to do this double-save thing, you can just do it all at once

Reply:
Yeah... I had a different error at first and it wasn't working and I couldn't catch the exception. I'll change back.